### ISSUE TITLE
Occasional 1-2 node startup tasks failing with no error message

### ISSUE THREAD

**Issue by lediur** [Azure/batch-shipyard#31]
Hey, I've started using the latest `develop` branch (4eea944), and every time I create a pool with 10 nodes, I inevitably get one or two nodes that fail to start properly now. I haven't had any fail to start before. The only messages I get from the startup task are:

in `startup/stdout.txt`:

```
/dev/sdb1 on /mnt type fuseblk (rw,relatime,user_id=0,group_id=0,allow_other,blksize=4096)
/dev/sdb1 temp disk is mounted as fuseblk/ntfs
```

`startup/stderr.txt` is empty.

My `pool.json` looks like:

```json
{
    "pool_specification": {
        "id": "mypool",
        "vm_size": "STANDARD_D11_V2",
        "vm_count": 10,
        "max_tasks_per_node": 2,

        "publisher": "Canonical",
        "offer": "UbuntuServer",
        "sku": "16.04.0-LTS",

        "ssh": {
            "username": "hi"
        },

        "reboot_on_start_task_failed": true,
        "block_until_all_global_resources_loaded": true
    }
}
```

With 5 nodes, I haven't run into any issues. Any idea what might be causing this?

**Comment by alfpark** [comment]
@lediur That stdout text corresponds to: https://github.com/Azure/batch-shipyard/blob/master/scripts/shipyard_nodeprep.sh#L136-L143

This is a lower-layer bug that has affected Ubuntu 16.04 distributions in the past. I might be able to add some more code to be aggressive about recovery, but this was *supposed* to be fixed between cloud-init and Azure's Linux agent.

**Comment by alfpark** [comment]
I decided to double check the image versions... looks like `16.04.0-LTS` latest stops at 2016-11-15 and the new `16.04-LTS` images have picked up since then. This explains why this issue is still happening although the fix has been in place for a while.

This will require changes to the Batch service to enable these images for deployment. I'll begin the process internally and update here when you can deploy `16.04-LTS` rather than `16.04.0-LTS`.

**Comment by lediur** [comment]
Okay, thanks. I'm trying out CentOS for now.

For other's reference, the `pool.json` VM image props I ended up using:

```json
{
    "pool_specification": {
        "publisher": "OpenLogic",
        "offer": "CentOS",
        "sku": "7.0",
    }
}
```

**Comment by alfpark** [comment]
You can issue the command `shipyard pool listskus` to list all of the available publisher/offer/sku combinations available for Azure Batch. CentOS 7.2 is available. We will be working on making 7.3 available (hopefully at the same time as the Ubuntu change).

**Comment by alfpark** [comment]
Fixed in 2.6.0b1. CentOS 7.3 is also available.
